crates/ra_ide_api/src/expand_macro.rs

```diff
@@ -0,0 +1,113 @@
+//! FIXME: write short doc here
+
+use crate::{db::RootDatabase, FilePosition};
+use hir::db::AstDatabase;
+use ra_db::SourceDatabase;
+use rustc_hash::FxHashMap;
+
+use ra_syntax::{
+    algo::{find_node_at_offset, replace_descendants},
+    ast::{self},
+    AstNode, NodeOrToken, SyntaxKind, SyntaxNode, WalkEvent,
+};
+
+fn insert_whitespaces(syn: SyntaxNode) -> String {
```

Comment:
I guess a simple useful extension here is to print newline and increase/decrease indent when seeing `{}`

Reply:
It would also be cool to share logic here and in the `mbe` tests, which are pretty unreadable at the moment. But this is nice-to-have

Reply:
I added some other whitespaces around some tokens and indents. And added a FIXME about the `mbe` tests too.